### PR TITLE
Suppress warning while insert completion

### PR DESCRIPTION
**Pull request by mattn**
shortmes option have `c` flag to suppress warning for insert completion. This is useful to return `[]` with no-errors in CompleteFunc. But when the plugin author want to suppress the warning without modifying this option, there is no way to do it. This change allow to return v:null in CompleteFunc (findstart=0).

This is useful to show completion lazy.

## Old behavior

old

## New behavior

terminal6

### CODE REVIEW

File: src/edit.c [hunk 1]
```diff
@@ -4247,6 +4248,9 @@ expand_by_function(
 	    case VAR_DICT:
 		matchdict = rettv.vval.v_dict;
 		break;
+	    case VAR_SPECIAL:
+		if (rettv.vval.v_number == VVAL_NONE)
+		    compl_opt_suppress_empty = TRUE;
```
Comment (h-east) [inline src/edit.c:4253]: Please choose either `break;` or `// FALLTHROUGH`.

File: src/testdir/test_ins_complete.vim [hunk 2]
```diff
@@ -142,7 +142,15 @@ func Test_completefunc_args()
   delfunc CompleteFunc
 endfunc
 
-func s:CompleteDone_CompleteFuncDict( findstart, base )
+function! s:CompleteDone_CompleteFuncNone( findstart, base )
```
Comment (dpelle) [inline src/testdir/test_ins_complete.vim]: Patch 8.1.0711 replaced `function!` into `func` (without exclamation
mark) in tests. So we should stay consistent with the new style now.

File: runtime/doc/insert.txt [hunk 3]
```diff
@@ -1076,6 +1076,10 @@ that contains the List.  The Dict can have these items:
 			The only value currently recognized is "always", the
 			effect is that the function is called whenever the
 			leading text is changed.
+
+If you want to suppress warning message for empty result, return v:none. This
```
Comment (h-east) [inline runtime/doc/insert.txt]: We need to write in the style of two spaces after the period.

File: runtime/doc/insert.txt [hunk 4]
```diff
@@ -1076,6 +1076,10 @@ that contains the List.  The Dict can have these items:
 			The only value currently recognized is "always", the
 			effect is that the function is called whenever the
 			leading text is changed.
+
+If you want to suppress warning message for empty result, return v:none.  This
+is useful to implement asynchronized completion with complete().
```
Comment (k-takata) [inline runtime/doc/insert.txt]: "asynchronous" might be better than "asynchronized"?

**Comment by brammool** [comment]
> shortmes option have `c` flag to suppress warning for insert
> completion. This is useful to return `[]` with no-errors in
> CompleteFunc. But when the plugin author want to suppress the warning
> without modifying this option, there is no way to do it. This change
> allow to return v:null in CompleteFunc (findstart=0).
> 
> This is useful to show completion lazy.
> 
> ## Old behavior
> 
> old
> 
> ## New behavior
> 
> terminal6

Thanks.

I notice the test passes even without the code change.  Would be nice if
it would fail without the extra code.

-- 
hundred-and-one symptoms of being an internet addict:
152. You find yourself falling for someone you've never seen or hardly
     know, but, boy can he/she TYPE!!!!!!

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by mattn** [comment]
Vim script can't know what messages are shown. Do you have any idea?

**Comment by brammool** [comment]
> Vim script can't know what messages are shown. Do you have any idea?

It should work using screenchar().  You can use the Screenline()
function from testdir/shared.vim

Another way is to use a screenshot test, but that makes running the test
a bit more complicated.

-- 
hundred-and-one symptoms of being an internet addict:
171. You invent another person and chat with yourself in empty chat rooms.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by mattn** [comment]
Ah, okay. will do in later.

**Pull request closed without merging**